src/templates/common/elements/pagination.html

```diff
@@ -0,0 +1,19 @@
+{% load pages %}
+
+<div class="pagination-block">
+    <ul class="pagination">
+        {% if page_obj.has_previous %}
+            <li class="arrow"><a href="{{ request.path }}?page={{ page_obj.previous_page_number }}">&laquo;</a></li>
+        {% endif %}
+
+        {% for page in page_obj|slice_pages:3 %}
+            <li class="{% if page_obj.number == page.number %}current{% endif %}">
+                <a href="?page={{ page.number }}">{{ page.number }}</a>
+            </li>
+        {% endfor %}
+
+        {% if page_obj.has_next %}
+            <li class="arrow"><a href="{{ request.path }}?page={{ page_obj.next_page_number }}">&raquo;</a></li>
+        {% endif %}
+    </ul>
```

Comment:
This is a usefull common element. One other thing I would add is a button for first and last page, which you can workout with `article_list.num_pages`.

In order to make it more re-usable, I would use a name like `paginated_results` and then include this element above with:
`{% include "common/elements/pagination.html"  with paginated_results=article_list%}`

Reply:
I've now added the button for first and last via a new template tag, `slice_pages_with_first_last_ellipsis`.

As for reusability, I'm not clear on how `paginated_results` that would make it more reusable. I don't refer to article_list in this template, just `page_obj` (and with the other commit this morning, `is_paginated` and `paginate_by`, all three of which are now passed by the CBV base class.

Maybe related--should this be reusable beyond CBVs?

Reply:
With your new template tag, my previous comment is irrelevant. I also like using the paginator rather than the queryset for consistency with what `slice_pages` used to do. Well done!